src/axom/primal/tests/primal_solid_angle.cpp

```diff
@@ -478,6 +478,195 @@ TEST(primal_solid_angle, selfintersecting_quadrilateral)
+                winding_number(queries[n], quad_patch, quad_tol, edge_tol, EPS),
+                1e-10);
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_integral, bezierpatch_sphere)
+{
+  using Point3D = primal::Point<double, 3>;
+  using Vector3D = primal::Vector<double, 3>;
+  using BPatch = primal::BezierPatch<double>;
+
+  double rt2 = sqrt(2), rt3 = sqrt(3), rt6 = sqrt(6);
+
+  // Define 6 rational bezier patches that define the surface of a sphere
+  // clang-format off
+  Point3D node_data[25] = {
+    Point3D {4*(1-rt3),     4*(1-rt3),     4*(1-rt3)}, Point3D {rt2*(rt3-4),            -rt2, rt2*(rt3-4)}, Point3D {4*(1-2*rt3)/3,   0, 4*(1-2*rt3)/3}, Point3D {rt2*(rt3-4),           rt2,   rt2*(rt3-4)}, Point3D {4*(1-rt3),     4*(rt3-1),     4*(1-rt3)},
+    Point3D {     -rt2, rt2*(rt3 - 4), rt2*(rt3 - 4)}, Point3D {(2-3*rt3)/2,     (2-3*rt3)/2,  -(rt3+6)/2}, Point3D {rt2*(2*rt3-7)/3, 0,      -5*rt6/3}, Point3D {(2-3*rt3)/2,   (3*rt3-2)/2,    -(rt3+6)/2}, Point3D {     -rt2,   rt2*(4-rt3),   rt2*(rt3-4)},
+    Point3D {        0, 4*(1-2*rt3)/3, 4*(1-2*rt3)/3}, Point3D {          0, rt2*(2*rt3-7)/3,    -5*rt6/3}, Point3D {0,               0,   4*(rt3-5)/3}, Point3D {          0, rt2*(7-2*rt3)/3,    -5*rt6/3}, Point3D {        0, 4*(2*rt3-1)/3, 4*(1-2*rt3)/3},
+    Point3D {      rt2, rt2*(rt3 - 4), rt2*(rt3 - 4)}, Point3D {(3*rt3-2)/2,     (2-3*rt3)/2,  -(rt3+6)/2}, Point3D {rt2*(7-2*rt3)/3, 0,      -5*rt6/3}, Point3D {(3*rt3-2)/2,   (3*rt3-2)/2,    -(rt3+6)/2}, Point3D {      rt2,   rt2*(4-rt3),   rt2*(rt3-4)},
+    Point3D {4*(rt3-1),     4*(1-rt3),     4*(1-rt3)}, Point3D {rt2*(4-rt3),            -rt2, rt2*(rt3-4)}, Point3D {4*(2*rt3-1)/3,   0, 4*(1-2*rt3)/3}, Point3D {rt2*(4-rt3),           rt2,   rt2*(rt3-4)}, Point3D {4*(rt3-1),     4*(rt3-1),     4*(1-rt3)}};
+
+  double weight_data[25] = {
+         4*(3-rt3), rt2*(3*rt3-2),   4*(5-rt3)/3, rt2*(3*rt3-2),     4*(3-rt3),
+     rt2*(3*rt3-2),     (rt3+6)/2, rt2*(rt3+6)/3,     (rt3+6)/2, rt2*(3*rt3-2),
+       4*(5-rt3)/3, rt2*(rt3+6)/3, 4*(5*rt3-1)/9, rt2*(rt3+6)/3,   4*(5-rt3)/3,
+     rt2*(3*rt3-2),     (rt3+6)/2, rt2*(rt3+6)/3,     (rt3+6)/2, rt2*(3*rt3-2),
+         4*(3-rt3), rt2*(3*rt3-2),   4*(5-rt3)/3, rt2*(3*rt3-2),     4*(3-rt3)
+  };
+  // clang-format on
+
+  BPatch sphere_faces[6];
+  for(int n = 0; n < 6; ++n)
+  {
+    sphere_faces[n].setOrder(4, 4);
+    sphere_faces[n].makeRational();
+  }
+
+  sphere_faces[0].setOrder(4, 4);
+  for(int i = 0; i < 5; ++i)
+  {
+    for(int j = 0; j < 5; ++j)
+    {
+      int idx = 5 * i + j;
+      for(int n = 0; n < 6; ++n)
+        sphere_faces[n].setWeight(i, j, weight_data[idx]);
+
+      // Set up each face
+      sphere_faces[0](i, j)[0] = node_data[idx][1] / weight_data[idx];
+      sphere_faces[0](i, j)[1] = node_data[idx][0] / weight_data[idx];
+      sphere_faces[0](i, j)[2] = node_data[idx][2] / weight_data[idx];
```

Comment:
Minor: I'm not sure if it will clean things up, but you should be able to set the points as `sphere_faces[-](i,j) = node_data[-]` and then divide the points by a scalar using the `.array()` function on the `primal::Point` class, e.g.   `pt.array() / = weight_data[-]`

(I used `-` as a placeholder for an actual index)

See: 
* https://github.com/LLNL/axom/blob/dd7ce1fca4540d04082e6c32c99a0041f41898b0/src/axom/primal/geometry/Point.hpp#L147-L154
* https://github.com/LLNL/axom/blob/dd7ce1fca4540d04082e6c32c99a0041f41898b0/src/axom/primal/geometry/NumericArray.hpp#L116-L124

Reply:
This does help it look a little more manageable, I think. The reason I have everything indexed with `idx` instead of `5*i + j` is just so that it all stays on one line, and isn't wrapped by the clang format. With `/=` handling the weights, it'll now fit if we expand it out, but I sort of think using `idx` looks neater.

Reply:
If we used the `axom::Array<Point3D,2>` we should be able to index it as `arr[i][j]`